xbmc/windowing/gbm/WinSystemGbm.cpp

```diff
@@ -85,6 +85,18 @@ bool CWinSystemGbm::InitWindowSystem()
       ->GetSettings()
       ->GetSetting("videoscreen.limitguisize")
       ->SetVisible(true);
+  CServiceBroker::GetSettingsComponent()
```

Comment:
can you please refactor this with the above code to something like:
```
auto settingsComponent = CServiceBroker::GetSettingsComponent();
if (!settingsComponent)
  return false;

auto settings = settingsComponent->->GetSettings();
if (!settings)
  return false;

auto setting = settings->GetSetting(CSettings::SETTING_VIDEOPLAYER_USEDISPLAYASCLOCK);
if (setting)
  setting->SetVisible(false);
```

Reply:
makes sense, will do!